sensor/common/scan/scan.go

```diff
@@ -297,10 +299,23 @@ func (s *LocalScan) getPullSources(srcImage *storage.ContainerImage) []*storage.
 			continue
 		}
 
+		// This ID assignment addresses an edge case where a podspec references an image
+		// by tag (e.g. latest) and the registry contents for that tag change after the pod was
+		// created. We want to ensure that we pull metadata and layers based on the
+		// ID (digest) of the running image instead of what the tag currently represents.
+		// This condition will only be true for mirrors setup via the ImageTagMirrorSet (ITMS) CR,
```

Comment:
is it possible the mirror messes with the digest, too? Perhaps out-of-scope for this PR, but asking out of curiosity? I don't think those CRs allow for that, but I'm curious if it's possible. Would be weird, though...

Reply:
Currently doesn't appear so based on scanning the library `c/Image` code which creates the mirror list - however the library is positioned in such a way where it could mess with the digest.

Can't think of a legit scenario where that would make sense - stretching here but perhaps if it is known that digests between registries will not match, then the library could be instructed to 'remove' the digest from the references in favor of some 'default tag'?